src/workflows/dags/pull__satellite_data.py

```diff
@@ -0,0 +1,72 @@
+from airflow.decorators import dag, task
+from datetime import datetime, timedelta
+import pytz
+from workflows.airqo_etl_utils.satellite_data_utils import DataHandler
+from airqo_etl_utils.workflows_custom_utils import AirflowUtils
+from datetime import datetime, timedelta
+import time
+import pytz
+
+@dag(
+    "satellite_data_processing",
+    schedule='0 0 */2 * *',
+    default_args=AirflowUtils.dag_default_configs(),
+    catchup=False,
+    tags=["airqo", "satellite"],
+)
+@task()
+def satellite_data_processing():
+    # Create an instance of DataHandler
+    data_handler = DataHandler()
+
+    # Example usage of query_bigquery_batch
+    start_time = datetime.now(tz=pytz.UTC) - timedelta(days=15)
+    end_time = datetime.now(tz=pytz.UTC)
+    batch_size = 5000
+    total_rows = 0
+    last_timestamp = None
+
+    while start_time <= end_time:
+        print("Querying BigQuery...")
+        data = data_handler.query_bigquery_batch(start_time=start_time, end_time=end_time, batch_size=batch_size)
+        if data.empty:
+            break
+        print("Querying BigQuery complete.")
+
+        print("Processing geolocation data...")
```

Comment:
@wabinyai its not a good practice to leave print statements in production codebases. if you wish to view logs, you can use the python logger, however airflow also does this by default (check out one of our production dags while its running. 
```suggestion
  
```
[here](https://medium.com/red-buffer/python-production-level-coding-practices-4c39246e0233#:~:text=Logging%20Critical%20Failures%20and%20Intermediate%20Results)

Reply:
You are right, the print was for testing if the line was running fine. These will be removed before it is merged into staging/production